### ISSUE TITLE
Single Sign On

### ISSUE THREAD

**Issue by camelspeed** [Project-Heimdall/home#115]
- [ ] Navigating to the application can only be achieved if authenticated by the enterprise SSO system, otherwise user shown standard enterprise error page.

**Comment by onetrickwolf** [comment]
Waiting on examples and info.

**Comment by camelspeed** [comment]
Java OAuth example shown so should not be blocked.  The JS version was for HTTP Rest but let's move with OAuth.  If still blocked let me know.

**Comment by camelspeed** [comment]
We have the right info moving forward.

**Comment by onetrickwolf** [comment]
PF server and PF integration kit concept proven locally. 

Connection to real SSO set up properly and working to the point that we get this error:

"Unexpected exception occurred in Response Handling: Redirect link doesn't match any expected values.
Looks like something is not right. Please contact csn_support@bah.com.
Partner: Stage_IDP
Target Resource: http://127.0.0.1/pf/index.html?cmd=PingStartPage"

This is expected I assume but we will need to set up some sort of live staging domain in order to test the SSO off of production. Submitted CR and see what the SSO team recommends.

Apache 2.2 and Ubuntu 12 was used to get the request properly through after having much issue with Apache 2.4 and Ubuntu 14/16. Am going to try to work up the configuration to at least Apache 2.4 and Ubuntu 14 now that I was able to verify it works.

**Comment by onetrickwolf** [comment]
Currently waiting on some way to confirm tests (most likely by deploying on AWS)

**Comment by onetrickwolf** [comment]
AWS box up installing PF and Apache on it for SSO test.

**Comment by camelspeed** [comment]
waiting on corp to open ips

**Comment by onetrickwolf** [comment]
SSO authenticating and getting user data back, testing user data integration on #152, will work with @camelspeed to integrate this in to dev environment

**Comment by onetrickwolf** [comment]
Documentation deliver to @camelspeed standing by to help deploy, once deployed will mark this as complete.

**Comment by onetrickwolf** [comment]
A couple of issues that may need to be their own issue, PF strips push states meaning direct links won't work (you are already redirected to the index). This may be solvable on our end but may need to consult with PF team.

Another issue is if you let the session time out, direct HTTP fetches from within the app cannot authenticate due to CORS issues. This may be something the PF team can enable but if not possible we can trigger a refresh if we detect a CORS failure. The auth is valid for ~5 mins without activity so this could be an issue.

**Comment by onetrickwolf** [comment]
Looks like SSO is working on dev and configs are cleaned up, moving this to review.
